### DISCUSSION TITLE
Why do we have CorrelationIdOperationFilter? app.UseCorrelationId() already sets a X-Correlation-ID header.

### DISCUSSION THREAD

**Discussion by VictorioBerra** [Dotnet-Boxed/Templates#215] [Q&A]
As per the docs, also I tested this myself and got this: `X-Correlation-ID: 0HLF2MM0MDN26:00000007`.

> The TraceIdentifier on the HttpContext will be used for new requests and additionally set a header on the response.

Poor wording but it does in fact set the header.

**Comment by RehanSaeed** [comment]
All `CorrelationIdOperationFilter` does is let you set the `X-Correlation-ID` from Swagger by providing a text box and some nice documentation.

The whole point of `X-Correlation-ID` is to send a unique GUID from your client to your server (if you control both), then include the `X-Correlation-ID` in any logs (I need to add this as a feature). This gives you the ability to correlate logs from your client to your server.
